pkg/util/util.go

```diff
@@ -49,7 +49,8 @@ const (
 )
 
 var (
-	rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
+	rnd          = rand.New(rand.NewSource(time.Now().UnixNano()))
+	ErrNoCluster = errors.Errorf("no %q label present", clusterv1.MachineClusterLabelName)
```

Comment:
I believe this call to `errors.Errorf` here will cause the error's stack trace information to trace back to the line, and not the `return` statement in `GetClusterFromMetadata`.

Reply:
That should only apply to Wrapf afaik

Reply:
It applies to Errorf too:

https://github.com/pkg/errors/blob/27936f6d90f9c8e1145f11ed52ffffbfdb9e0af7/errors.go#L109-L111

Reply:
I'll switch to fmt.Errorf here

Reply:
👍 

and return `errors.WithStack` below?

Reply:
I assume given that this is a library, callers would wrap the errors instead, wdyt?

Reply:
Callers can wrap too, but if we're generating an error, it should have stack info.

Reply:
Done